RNSound/RNSound.m

```diff
@@ -241,7 +241,12 @@ - (NSDictionary *)constantsToExport {
         addObserver:self
            selector:@selector(audioSessionChangeObserver:)
                name:AVAudioSessionRouteChangeNotification
-             object:nil];
+             object:[AVAudioSession sharedInstance]];
+    [[NSNotificationCenter defaultCenter]
+        addObserver:self
+           selector:@selector(audioSessionChangeObserver:)
+               name:AVAudioSessionInterruptionNotification
+             object:[AVAudioSession sharedInstance]];
```

Comment:
We also need to listen to the `AVAudioSessionInterruptionNotification` event such that the handler is called as expected.